pkg/registry/client.go

```diff
@@ -0,0 +1,177 @@
+package registry
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strings"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/v2"
+	orasregistry "oras.land/oras-go/v2/registry"
+	"oras.land/oras-go/v2/registry/remote"
+	"oras.land/oras-go/v2/registry/remote/auth"
+)
+
+// OCIRegistryClient storage client for an OCI registry.
+type OCIRegistryClient struct {
+	host        string
+	project     string
+	certFile    string
+	insecure    bool
+	dryRun      bool
+	initialized bool
+	OI          OrasInterface
+	registry    *remote.Registry
+}
+
+var _ StorageClient = (*OCIRegistryClient)(nil)
+
+// NewOCIRegistry create an OCI registry client.
+func NewOCIRegistry(host string, certFile string, insecure bool) *OCIRegistryClient {
+	return &OCIRegistryClient{
+		host:     host,
+		certFile: certFile,
+		insecure: insecure,
+		OI:       &OrasImplementation{},
+	}
+}
+
+// Init registry configuration.
+func (or *OCIRegistryClient) Init() error {
+	if or.initialized {
+		return nil
+	}
+
+	credentialStore := NewCredentialStore()
```

Comment:
For the future, it's typically a better pattern to build these in your main, and pass them to a constructor. That makes your code more testable (you can easily pass in a test implementation), and it corrals the possible Init() errors to the main method where they can more easily be dealt with.